src/main/app/fees/feesClient.ts

```diff
@@ -34,49 +35,48 @@ export class FeesClient {
    * @returns {Promise.<number>} promise containing the fee amount in pounds
    */
   static async calculateHearingFee (claimValue: number): Promise<number> {
-    const fee: CalculationOutcome = await this.calculateFee(hearingFeeCode, claimValue)
-    return MoneyConverter.convertPenniesToPounds(fee.amount)
+    return this.calculateFee(hearingFeeEvent, claimValue, defaultChannel)
+      .then((outcome: FeeOutcome) => (outcome.amount))
   }
-
   /**
-   * Calculates the fee based on fee code and amount
+   * Calculates the fee based on fee event and amount
    *
-   * @param feeCode which fee category to use
+   * @param eventType which fee event to use
    * @param amount amount in pounds
-   * @returns {Promise.<CalculationOutcome>} promise containing the calculation outcome (including fee amount in pennies)
+   * @returns {Promise.<FeeOutcome>} promise containing the Fee outcome (including fee amount in GBP)
    */
-  static async calculateFee (feeCode: string, amount: number): Promise<CalculationOutcome> {
-    ClaimValidator.claimAmount(amount)
-    const amountInPennies = MoneyConverter.convertPoundsToPennies(amount)
-    if (amountInPennies <= 0) {
-      throw new Error(`Amount must be at least 1 penny, amount was: ${amountInPennies}`)
+  static async calculateFee (eventType: string, amount: number, channel: string): Promise<FeeOutcome> {
+    if (StringUtils.isBlank(eventType)) {
```

Comment:
Please add similar preconditions for other arguments like channel and amount.

Reply:
I think all of the review comments are actioned. Pls have a look when you've a moment.